### ISSUE TITLE
bq 4.5 signal / textsecure is starting & crashing

### ISSUE THREAD

**Issue by bakuny** [nanu-c/textsecure-qml#42]
### Description

signal starts as expected but it isn't possible to write a message (no button or keyboard pops up), to receive a message. after importing contacts or linking a device it crashes.

**Reproduces how often:** [What percentage of the time does it reproduce?]
100%

#### Versions
0.3.261

#### Device
bq aquaris 4.5 ubuntu edition, ubuntu touch 16.04 (ota-5)

**Comment by sinnumero** [comment]
Same issue, but with no more crashing after deleteing all files related the app and then
doing a re-installation. I did some investigations and can add the following:

- contact import stores them in ~/.config/textsecure.nanuc/contacts.yml, but nothing shows up,
  stating "no contacts". I checked ownership and permissions of the file: looks correct. But
  I could not find any open(2) system call on that file or a similar one using `strace -f -e open -p ...`
- After receiving signal messages, these appear correctly and one can answer to them as
  expected.
I'm going to save the contact's files, one from the address book import and anoher one
after receving messages; maybe I find some hints there.

Version:  0.3.261
Device:   Aquaris 4.5 ubuntu edition, ubuntu touch 15.04 (vivid vervet)  

(unfortunately, I also could not get it running in the libertine container up to now)

**Comment by nanu-c** [comment]
It's really had to reproduce this error. I need more investigations. Someone reported 401 errors while importing the phonebook

**Comment by Danfro** [comment]
Can't confirm on E5 with 16.04 with latest RC.

But contacts are shown only with network connection available.
If they are stored in a file, why is that?

**Comment by Beechm** [comment]
My nexus 5 Running UT16.04 has the same issue with this app. there is no way to start a message or add contacts. I was able to link my phone number to the app and link the app to the desktop app, but that is all I can do with this. Is someone working on a fix? Will this be sorted out in the next update?

**Comment by Danfro** [comment]
I imported contacts from addressbook yesterday without any problems. E5 with RC using signal app 0.4.1.

**Comment by Beechm** [comment]
There is one more characteristic to this bug I noticed. The pencil icon shown in previews of the app on appstore does not appear on the app loaded on either of my Nexus5 phones. I assume this icon links to a field where you can enter a number or add people to contact list.

**Comment by milkbubi** [comment]
I have a Nexus 5 with current Ubuntu 16. 04 (OTA-10). Contacts cannot be imported. Signal Version is 0.4.4.

Is there a way to use the App ?

**Comment by Danfro** [comment]
@milkbubi are you using uAdblock with propaganda & fakenews category filtered? then remove that filter and see if that helps.

**Comment by milkbubi** [comment]
@Danfro Thanks for the quick answer. I have removed the uAdblock but unfortunately the problem could not be solved this way. . .

**Comment by Danfro** [comment]
Are you using the latest version (0.4.4)? There has been something fixed regarding contact import too. After that update and with uAdblock turned off it was working again for me.

**Comment by Danfro** [comment]
@milkbubi  And try to clear the cache and maybe do need a reboot after resetting uAdblock lists.

**Comment by milkbubi** [comment]
@Danfro 
i rebooted the device but it still doesn't work. i installed a current Ubuntu touch today and re-entered the contacts on the smartphone. is there maybe another possibility?

**Comment by Danfro** [comment]
@milkbubi  you can export and import .vcf files.

Copy your contacts.vcf file onto the device, open that file with filemanager and select contacts app to import them. In contacts app you can export contacts to vcf via filemanager.

**Comment by milkbubi** [comment]
@danfro I have the contacts as vcf generated and read in with the contacts app. It still does not work. Thank you for your help!

**Comment by milkbubi** [comment]
since the last update this morning I can add the contacts. Unfortunately you can't start new chats or receive single messages unless you create a new group. . .

**Comment by nanu-c** [comment]
Lot have been done on the contact import, please test v0.7.0 and report upcoming issues again.

**Comment by soki3456** [comment]
Installed today on nexus 5 - no contacts can be added: nor though the dialog field, after entering and confirming, the contact isn't listed anywhere, nor does the import work, i just tells me to com back later, but then no contact is in the list.

**Comment by nanu-c** [comment]
Do you have contacts that are registered with signal? Because if not the list is empty. It works like, hash your telephone numbers from your contacts, ask signal server if they exist, and if yes they are displayed. Note this is done asynchronous and takes a while. if you quit axolotl, its interrupted.
This is also done, when you restart axolotl.
So to support you, try to add a contact manually from a phone number you know, that is connected with signal, if that doesn't work please open a new issue and attach the log. Thanks

**Comment by soki3456** [comment]
i tried manual import svereral time - still no contacts listed. 
sorry - where do I find a manual to find the logfile, I'd like to share. Thanks a lot.

**Comment by Danfro** [comment]
@soki3456 For logs you can use either [Logviewer app](https://open-store.io/app/logviewer.neothethird) and send the log to pastebin with pasting the link here or check the logs on your device under `.cache/upstart/`.
